### PR TITLE
[WIP] feat: allow using big number when creating/updating an artwork in MyC

### PR DESCRIPTION
**Pull request by dariakoko**
When we call Create or Update Artwork mutation and the artwork price that gets sent in this request is too high, the request fails. It happens because we used `GraphQLInt` and it doesn't support big numbers

I'm using GraphQLFloat here, but we can also use a custom type as in [this commit](https://github.com/artsy/metaphysics/pull/4556/commits/3f262a8ae27b759ec732daa00854adee05e74fc2) (thanks @lordkiz!). I'm using float in the end because the custom type requires way more changes and not only on the create/update artwork mutations

### CODE REVIEW

**Review by MounirDhahri — CHANGES_REQUESTED** [review]
I think the changes in this file will be breaking old client apps and I believe we should be handling this differently

File: src/schema/v2/me/myCollectionCreateArtworkMutation.ts [hunk 1]
```diff
@@ -115,7 +116,7 @@ export const myCollectionCreateArtworkMutation = mutationWithClientMutationId<
       type: GraphQLString,
     },
     pricePaidCents: {
```
Comment (dblandin) [inline src/schema/v2/me/myCollectionCreateArtworkMutation.ts:118]: There's a costMinor field further up in this file that is still represented as a GraphQLInt. Should that be updated as well?

File: src/schema/v2/fields/money.ts [hunk 2]
```diff
@@ -159,7 +159,7 @@ export const Money = new GraphQLObjectType<any, ResolverContext>({
   name: "Money",
   fields: {
     minor: {
-      type: new GraphQLNonNull(GraphQLInt),
+      type: new GraphQLNonNull(GraphQLFloat),
```
[the remark this answers is not in the document]
Comment (dblandin) [reply src/schema/v2/fields/money.ts:162]: Is this field in the shared Money type used in the My Collection flow?

**Review by dblandin — COMMENTED** [review]
@MounirDhahri There is some precedent for upgrading fields like this in-place: https://github.com/artsy/metaphysics/pull/969

Any GraphQLInt input should be compatible with a GraphQLFloat input arg type. So this might not cause any problems downstream but we should QA by running Force or Eigen against a locally running copy of Metaphysics or merge and place a deploy block on Metaphysics which QA takes place on staging.

**Comment by MounirDhahri** [comment]
@dblandin thanks for the comments. We spoke about this today and Kizito also mentioned that this is possible and he have tested it with @dariakoko.
We spoke also about the fact that this update will be relfected in all money fields in Artsy, and not just MyCollection flow, that's why, what we will do is to use the major and minor fields as ints and this will allow us to accept numbers up to 2 billions which should be more than enough for my collection artworks

**Comment by mzikherman** [comment]
Bump! (not sure if this is still relevant, if not can be closed)

**Comment by lordkiz** [comment]
This has been fixed in https://github.com/artsy/metaphysics/pull/4709 and https://github.com/artsy/metaphysics/pull/4702

**Pull request closed without merging**